src/main/java/org/jenkinsci/plugins/workflow/cps/GroovyShellDecorator.java

```diff
@@ -38,7 +38,26 @@ public void configureCompiler(@CheckForNull CpsFlowExecution context, CompilerCo
      */
     public void configureShell(@CheckForNull CpsFlowExecution context, GroovyShell shell) {}
 
+    /**
+     * Obtains a contextualized {@link GroovyShellDecorator} used to decorate the trusted shell.
+     *
+     * <p>
+     * By default, this method returns null decorator that doesn't do anything.
```

Comment:
Previously there was no trusted classloader, so old plugins like docker-workflow that implements `GroovyShellDecorator` will continue to insert classpath into the untrusted classloader. So I think this is compatible.

I'll run the docker workflow test to verify.